packages/dara-core/js/api/websocket.tsx

```diff
@@ -302,6 +302,8 @@ export class WebSocketClient implements WebSocketClientInterface {
                 }
             });
         });
+        // Rebind the close handler so this logic is re-applied on a subsequent disconnect
+        this.socket.addEventListener('close', this.closeHandler);
```

Comment:
Should we extract logic from the constructor and simply call it again here? Seems like the duplication is a bit brittle and can lead to more mistakes